tests/monitoring_integration_test.py

```diff
@@ -106,6 +106,28 @@ def testUserExpDashboard(self):
 				
 		self.assertTrue(found_userexp_dashboard)
 
+	def testAdserviceDashboard(self):
+		""" Test that the Adservice Dashboard gets created. """
+		client = v1.DashboardsServiceClient()
```

Comment:
Depending on how many dashboards you end up creating/testing, it may be worth pulling this into a separate helper method since the only thing that changes is the display_name.

Reply:
So I refactored the test to test all the dashboards in the same test by iterating through the display names... The only issue with what I changed it to is it's hard to pin down which dashboard has the failure if it does fail. What do you think?

When you meant helper method did you mean still having one test case per dashboard but they all call a common function to do some of the extra work?

Reply:
Yes, sorry that wasn't clear. I think the block of code used to fetch all dashboards, and look for a specific display_name could have been extracted into its own method. Then you would call that from each test. You are right that having individual tests makes it easier to identify failures.

For example, in one test you could call `checkForDashboard('User Experience Dashboard')`

Reply:
Okay that makes a lot of sense. I've adjusted it to have individual tests for each dashboard, but they call the same helper function to do the bulk of the work.